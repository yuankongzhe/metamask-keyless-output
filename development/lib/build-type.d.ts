import { Infer, Struct } from 'superstruct';

export type Unique<Element extends Struct<any>> = (
  struct: Struct<Infer<Element>[], Infer<Element>>,
  eq?: (a: Infer<Element>, b: Infer<Element>) => boolean,
) => Struct<Infer<Element>[], Infer<Element>>;
