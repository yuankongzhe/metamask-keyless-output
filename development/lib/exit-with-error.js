/**
 * Exit the process with an error message.
 *
 * Note that this should be called before the process ends, but it will not
 * itself end the process. This is because the Node.js documentation strongly
 * advises against calling `process.exit` directly.
 *
 * @param {string} errorMessage - The error message that is causing the non-
 *   zero exit code.
 */
function exitWithError(errorMessage) {
  console.error(errorMessage);
  process.exitCode = 1;
  if (process.env.CI) {
    // For some reason, the CI environment does not exit when process.exitCode
    // is set, or at least doesn't terminate immediately. Something may be
    // keeping the event loop from ending. The result is that sometimes jobs
    // fail because of timeout even though none of the individual tests fail.
    // This is a workaround.
    process.exit();
  }
}

module.exports = { exitWithError };
