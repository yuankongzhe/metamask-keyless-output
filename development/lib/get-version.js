const { version: manifestVersion } = require('../../package.json');
const { loadBuildTypesConfig } = require('./build-type');

/**
 * Get the current version of the MetaMask extension. The base manifest version
 * is modified according to the build type and version.
 *
 * The build version is needed because certain build types (such as beta) may
 * be released multiple times during the release process.
 *
 * @param {string} buildType - The build type.
 * @param {number} buildVersion - The build version.
 * @returns {string} The MetaMask extension version.
 */
function getVersion(buildType, buildVersion) {
  return loadBuildTypesConfig().buildTypes[buildType].isPrerelease === true
    ? `${manifestVersion}-${buildType}.${buildVersion}`
    : manifestVersion;
}

module.exports = { getVersion };
