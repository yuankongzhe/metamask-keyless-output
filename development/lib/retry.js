/**
 * Re-runs the given function until it returns a resolved promise or the number
 * of retries is exceeded, whichever comes first (with an optional delay in
 * between retries).
 *
 * @param {object} args - A set of arguments and options.
 * @param {number} args.retries - The maximum number of times to re-run the
 * function on failure.
 * @param {number} [args.delay] - The amount of time (in milliseconds) to wait in
 * between retries. (Default: 0)
 * @param {string} [args.rejectionMessage] - The message for the rejected promise
 * this function will return in the event of failure. (Default: "Retry limit
 * reached")
 * @param {boolean} [args.retryUntilFailure] - Retries until the function fails.
 * @param {Function} functionToRetry - The function that is run and tested for
 * failure.
 * @returns {Promise<* | null | Error>} a promise that either resolves with one of the following:
 * - If successful, resolves with the return value of functionToRetry.
 * - If functionToRetry fails while retryUntilFailure is true, resolves with null.
 * - Otherwise it is rejected with rejectionMessage.
 */
async function retry(
  {
    retries,
    delay = 0,
    rejectionMessage = 'Retry limit reached',
    retryUntilFailure = false,
  },
  functionToRetry,
) {
  let attempts = 0;
  while (attempts <= retries) {
    if (attempts > 0 && delay > 0) {
      await new Promise((resolve) => setTimeout(resolve, delay));
    }

    try {
      const result = await functionToRetry();
      if (!retryUntilFailure) {
        return result;
      }
    } catch (error) {
      if (error.message === "Exited with code '1'") {
        console.log("retry() received: Exited with code '1'");
      } else {
        console.error('error caught in retry():', error);
      }

      if (attempts < retries) {
        console.log('Ready to retry() again');
      }

      if (retryUntilFailure) {
        return null;
      }
    } finally {
      attempts += 1;
    }
  }

  throw new Error(rejectionMessage);
}

module.exports = { retry };
