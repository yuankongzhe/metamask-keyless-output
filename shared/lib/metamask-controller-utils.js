export function getTokenValueParam(tokenData = {}) {
  return tokenData?.args?._value?.toString();
}
