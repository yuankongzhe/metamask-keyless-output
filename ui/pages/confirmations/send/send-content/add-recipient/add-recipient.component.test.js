import React from 'react';
import configureMockStore from 'redux-mock-store';
import { renderWithProvider } from '../../../../../../test/lib/render-helpers';
import mockState from '../../../../../../test/data/mock-state.json';
import mockSendState from '../../../../../../test/data/mock-send-state.json';
import AddRecipient from '.';

describe('Add Recipient Component', () => {
  describe('render', () => {
    const mockStore = configureMockStore()(mockState);
    it('should match snapshot', () => {
      const { container } = renderWithProvider(<AddRecipient />, mockStore);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Send State', () => {
    const mockStore = configureMockStore()(mockSendState);

    it('should match snapshot', () => {
      const { container } = renderWithProvider(<AddRecipient />, mockStore);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Domain Resolution', () => {
    const mockDomainResolutionState = {
      ...mockState,
      DNS: {
        resolution: 'DNS Resolution',
      },
    };
    const mockStore = configureMockStore()(mockDomainResolutionState);

    it('should match snapshot', () => {
      const { container } = renderWithProvider(<AddRecipient />, mockStore);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Own Account Recipient Search', () => {
    const ownAccountSeachState = {
      ...mockState,
      send: {
        ...mockState.send,
        recipientInput: 'Test',
        recipientMode: 'MY_ACCOUNTS',
      },
    };
    const mockStore = configureMockStore()(ownAccountSeachState);

    it('should match snapshot', () => {
      const { container } = renderWithProvider(<AddRecipient />, mockStore);

      expect(container).toMatchSnapshot();
    });
  });
});
