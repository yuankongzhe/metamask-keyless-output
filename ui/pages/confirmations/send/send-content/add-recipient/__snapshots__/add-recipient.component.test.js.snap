// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipient Component Domain Resolution should match snapshot 1`] = `
<div>
  <div
    class="box send__select-recipient-wrapper box--flex-direction-row"
  >
    <div
      class="box box--margin-top-2 box--margin-right-4 box--margin-bottom-2 box--margin-left-4 box--flex-direction-row"
    >
      <p
        class="mm-box mm-text mm-text--body-lg-medium mm-box--color-text-alternative"
      >
        Your accounts
      </p>
    </div>
    <div
      class="send__select-recipient-wrapper__list"
    >
      <div
        class="box send__select-recipient-wrapper__group box--flex-direction-row"
        data-testid="recipient-group"
      >
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(250, 58, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#18CDF2"
                    height="28"
                    transform="translate(-0.9173443158196976 -2.8912485607779366) rotate(328.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#035E56"
                    height="28"
                    transform="translate(-16.01115399522804 9.268404137780026) rotate(176.2 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F26602"
                    height="28"
                    transform="translate(14.584361765945934 -12.429497257622447) rotate(468.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0x0dcd...e7bc
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(245, 143, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#018E74"
                    height="28"
                    transform="translate(5.524091896954402 -4.749745570944453) rotate(358.4 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#18CAF2"
                    height="28"
                    transform="translate(6.724913619395038 -10.621076649467508) rotate(414.8 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#C81474"
                    height="28"
                    transform="translate(-4.907795154940259 -21.945972385956175) rotate(322.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 2
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xec1a...251b
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(249, 100, 1);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#C81432"
                    height="28"
                    transform="translate(-0.510690249475231 -2.7159619403490534) rotate(331.7 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#187AF2"
                    height="28"
                    transform="translate(-11.127931122273568 5.54021528290869) rotate(160.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#FB183E"
                    height="28"
                    transform="translate(25.084575445435807 -2.611206239472919) rotate(517.6 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Ledger Hardware 2
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xc42e...8813
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(242, 206, 2);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#2340E1"
                    height="28"
                    transform="translate(-1.7054541409645272 -0.893781972900803) rotate(257.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#016D8E"
                    height="28"
                    transform="translate(11.213543194590896 2.66257866392058) rotate(77.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F5F500"
                    height="28"
                    transform="translate(11.486654880888956 -19.46641853277226) rotate(465.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 3
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xeb9e...4823
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(24, 125, 242);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#C81435"
                    height="28"
                    transform="translate(-1.8544047661519008 -1.6499475479241468) rotate(249.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F5E400"
                    height="28"
                    transform="translate(-1.3241332948360764 10.505504944881555) rotate(269.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F96001"
                    height="28"
                    transform="translate(6.630832321372173 21.724427796185292) rotate(216.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Snap Account 1
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xb552...ee5d
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(250, 62, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#238FE1"
                    height="28"
                    transform="translate(-3.696976537625548 4.334010253908054) rotate(161.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#FB1891"
                    height="28"
                    transform="translate(-3.8726006416348677 11.341987744107225) rotate(224.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F90901"
                    height="28"
                    transform="translate(-23.479125860782805 -13.386102528655465) rotate(344.7 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 4
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xca8f...f281
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="send__select-recipient-wrapper__group-item"
    >
      <div
        class=""
      >
        <div
          class="identicon"
          style="height: 28px; width: 28px; border-radius: 14px;"
        >
          <div
            style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(24, 151, 242);"
          >
            <svg
              height="28"
              width="28"
              x="0"
              y="0"
            >
              <rect
                fill="#2362E1"
                height="28"
                transform="translate(3.140663911693453 -4.667521795329333) rotate(458.4 14 14)"
                width="28"
                x="0"
                y="0"
              />
              <rect
                fill="#F94301"
                height="28"
                transform="translate(-13.443026878443934 6.993307681050803) rotate(268.8 14 14)"
                width="28"
                x="0"
                y="0"
              />
              <rect
                fill="#FA7900"
                height="28"
                transform="translate(-7.941722319388745 25.795013895597954) rotate(117.3 14 14)"
                width="28"
                x="0"
                y="0"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="send__select-recipient-wrapper__group-item__content"
      >
        <div
          class="send__select-recipient-wrapper__group-item__title"
        >
          DNS Re...tion
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Add Recipient Component Own Account Recipient Search should match snapshot 1`] = `
<div>
  <div
    class="box send__select-recipient-wrapper box--flex-direction-row"
  >
    <div
      class="send__select-recipient-wrapper__list"
    >
      <div
        class="send__select-recipient-wrapper__list"
      >
        <div
          class="send__select-recipient-wrapper__recent-group-wrapper"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Add Recipient Component Send State should match snapshot 1`] = `
<div>
  <div
    class="box send__select-recipient-wrapper box--flex-direction-row"
  >
    <div
      class="send__select-recipient-wrapper__group-item"
    >
      <div
        class=""
      >
        <div
          class="identicon"
          style="height: 28px; width: 28px; border-radius: 14px;"
        >
          <div
            style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(245, 143, 0);"
          >
            <svg
              height="28"
              width="28"
              x="0"
              y="0"
            >
              <rect
                fill="#018E74"
                height="28"
                transform="translate(5.524091896954402 -4.749745570944453) rotate(358.4 14 14)"
                width="28"
                x="0"
                y="0"
              />
              <rect
                fill="#18CAF2"
                height="28"
                transform="translate(6.724913619395038 -10.621076649467508) rotate(414.8 14 14)"
                width="28"
                x="0"
                y="0"
              />
              <rect
                fill="#C81474"
                height="28"
                transform="translate(-4.907795154940259 -21.945972385956175) rotate(322.3 14 14)"
                width="28"
                x="0"
                y="0"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="send__select-recipient-wrapper__group-item__content"
      >
        <div
          class="send__select-recipient-wrapper__group-item__title"
        >
          A
          c
          c
          o
          u
          n
          t
           
          2
        </div>
        <div
          class="send__select-recipient-wrapper__group-item__subtitle"
        >
          0xec1a...251b
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Add Recipient Component render should match snapshot 1`] = `
<div>
  <div
    class="box send__select-recipient-wrapper box--flex-direction-row"
  >
    <div
      class="box box--margin-top-2 box--margin-right-4 box--margin-bottom-2 box--margin-left-4 box--flex-direction-row"
    >
      <p
        class="mm-box mm-text mm-text--body-lg-medium mm-box--color-text-alternative"
      >
        Your accounts
      </p>
    </div>
    <div
      class="send__select-recipient-wrapper__list"
    >
      <div
        class="box send__select-recipient-wrapper__group box--flex-direction-row"
        data-testid="recipient-group"
      >
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(250, 58, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#18CDF2"
                    height="28"
                    transform="translate(-0.9173443158196976 -2.8912485607779366) rotate(328.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#035E56"
                    height="28"
                    transform="translate(-16.01115399522804 9.268404137780026) rotate(176.2 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F26602"
                    height="28"
                    transform="translate(14.584361765945934 -12.429497257622447) rotate(468.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0x0dcd...e7bc
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(245, 143, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#018E74"
                    height="28"
                    transform="translate(5.524091896954402 -4.749745570944453) rotate(358.4 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#18CAF2"
                    height="28"
                    transform="translate(6.724913619395038 -10.621076649467508) rotate(414.8 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#C81474"
                    height="28"
                    transform="translate(-4.907795154940259 -21.945972385956175) rotate(322.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 2
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xec1a...251b
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(249, 100, 1);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#C81432"
                    height="28"
                    transform="translate(-0.510690249475231 -2.7159619403490534) rotate(331.7 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#187AF2"
                    height="28"
                    transform="translate(-11.127931122273568 5.54021528290869) rotate(160.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#FB183E"
                    height="28"
                    transform="translate(25.084575445435807 -2.611206239472919) rotate(517.6 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Ledger Hardware 2
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xc42e...8813
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(242, 206, 2);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#2340E1"
                    height="28"
                    transform="translate(-1.7054541409645272 -0.893781972900803) rotate(257.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#016D8E"
                    height="28"
                    transform="translate(11.213543194590896 2.66257866392058) rotate(77.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F5F500"
                    height="28"
                    transform="translate(11.486654880888956 -19.46641853277226) rotate(465.0 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 3
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xeb9e...4823
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(24, 125, 242);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#C81435"
                    height="28"
                    transform="translate(-1.8544047661519008 -1.6499475479241468) rotate(249.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F5E400"
                    height="28"
                    transform="translate(-1.3241332948360764 10.505504944881555) rotate(269.5 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F96001"
                    height="28"
                    transform="translate(6.630832321372173 21.724427796185292) rotate(216.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Snap Account 1
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xb552...ee5d
            </p>
          </div>
        </div>
        <div
          class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 28px; width: 28px; border-radius: 14px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(250, 62, 0);"
              >
                <svg
                  height="28"
                  width="28"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#238FE1"
                    height="28"
                    transform="translate(-3.696976537625548 4.334010253908054) rotate(161.9 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#FB1891"
                    height="28"
                    transform="translate(-3.8726006416348677 11.341987744107225) rotate(224.3 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F90901"
                    height="28"
                    transform="translate(-23.479125860782805 -13.386102528655465) rotate(344.7 14 14)"
                    width="28"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
            data-testid="recipient"
          >
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
            >
              Test Account 4
            </p>
            <p
              class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
            >
              0xca8f...f281
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="send__select-recipient-wrapper__list"
    >
      <div
        class="box box--margin-top-2 box--margin-right-4 box--margin-bottom-2 box--margin-left-4 box--flex-direction-row"
      >
        <p
          class="mm-box mm-text mm-text--body-lg-medium mm-box--color-text-alternative"
        >
          Contacts
        </p>
      </div>
      <div
        class="send__select-recipient-wrapper__list"
      >
        <div
          class="send__select-recipient-wrapper__recent-group-wrapper"
        />
        <div
          class="box send__select-recipient-wrapper__group box--flex-direction-row"
          data-testid="recipient-group"
        >
          <div
            class="box send__select-recipient-wrapper__group-label box--margin-top-2 box--margin-right-4 box--margin-bottom-2 box--margin-left-4 box--flex-direction-row"
          >
            <p
              class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            >
              A
            </p>
          </div>
          <div
            class="box send__select-recipient-wrapper__group-item box--padding-4 box--flex-direction-row"
          >
            <div
              class=""
            >
              <div
                class="identicon"
                style="height: 28px; width: 28px; border-radius: 14px;"
              >
                <div
                  style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 28px; height: 28px; display: inline-block; background: rgb(249, 100, 1);"
                >
                  <svg
                    height="28"
                    width="28"
                    x="0"
                    y="0"
                  >
                    <rect
                      fill="#C81432"
                      height="28"
                      transform="translate(-0.510690249475231 -2.7159619403490534) rotate(331.7 14 14)"
                      width="28"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#187AF2"
                      height="28"
                      transform="translate(-11.127931122273568 5.54021528290869) rotate(160.5 14 14)"
                      width="28"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#FB183E"
                      height="28"
                      transform="translate(25.084575445435807 -2.611206239472919) rotate(517.6 14 14)"
                      width="28"
                      x="0"
                      y="0"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="box send__select-recipient-wrapper__group-item__content box--flex-direction-row"
              data-testid="recipient"
            >
              <p
                class="mm-box mm-text send__select-recipient-wrapper__group-item__title mm-text--body-lg-medium mm-box--color-text-default"
              >
                Address Book Account 1
              </p>
              <p
                class="mm-box mm-text send__select-recipient-wrapper__group-item__subtitle mm-text--body-md mm-box--color-text-alternative"
              >
                0xc42e...8813
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
